src/pages/Auth/Auth.tsx

```diff
@@ -1,5 +1,93 @@
+import { useEffect } from 'react';
+import { useGoogleLogin, googleLogout } from '@react-oauth/google';
+import axios from 'axios';
+
+import { useAppDispatch } from '../../store/hooks';
+import { setUserInfo, setUserToken } from './authSlice';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../store';
+
+import styles from './style.module.scss';
+
+export const USER_TOKEN_KEY = 'user_token';
+const localToken = localStorage.getItem(USER_TOKEN_KEY);
+
+interface ResponseValue {
+  access_token: string;
+}
+
+interface UserInfo {
+  email: string;
+  name: string;
+  picture: string;
+}
+
 const Auth = () => {
-  return <div>AUTH PAGE</div>;
+  const dispatch = useAppDispatch();
+  const userToken =
+    useSelector((state: RootState) => state.auth.user_token) || localToken;
+  const { name, email, picture } = useSelector(
+    (state: RootState) => state.auth,
+  );
+
+  const login = useGoogleLogin({
+    onSuccess: (res: ResponseValue) => {
+      localStorage.setItem(USER_TOKEN_KEY, res.access_token);
+      dispatch(setUserToken(res.access_token));
+    },
+    onError: (error) => {
+      throw new Error(`Login Failed: ${error}`);
+    },
+  });
+
+  const logOut = () => {
+    googleLogout();
+    localStorage.removeItem(USER_TOKEN_KEY);
+    dispatch(setUserInfo(null));
+  };
+
+  useEffect(() => {
+    if (userToken) {
+      axios
+        .get(
+          `https://www.googleapis.com/oauth2/v1/userinfo?access_token=${userToken}`,
+          {
+            headers: {
+              Authorization: `Bearer ${userToken}`,
+              Accept: 'application/json',
+            },
+          },
+        )
+        .then((res) => {
+          const { name, email, picture }: UserInfo = res.data;
+
+          dispatch(setUserInfo({ name, email, picture }));
+          dispatch(setUserToken(localToken));
+        })
+        .catch((err) => {
+          throw new Error(err);
+        });
+    }
+  }, [userToken]);
```

Comment:
make a function outside of useEffect called `getUserInfo`. make it async. Move all logic there.
Also why catch error and then throw it back? Whats the purpose? It does nothing, better to just log it to console.
```suggestion
  const getUserInfo = async () => {
    try {
      const { data } = await axios
        .get(
          `https://www.googleapis.com/oauth2/v1/userinfo?access_token=${userToken}`,
          {
            headers: {
              Authorization: `Bearer ${userToken}`,
              Accept: 'application/json',
            },
          },
        );
        
      const { name, email, picture }: UserInfo = data;

      dispatch(setUserInfo({ name, email, picture }));
      dispatch(setUserToken(localToken));
      } catch (e) {
         console.log(e);
      }
  }

  useEffect(() => {
    if (userToken) {
      getUserInfo();
    }
  }, [userToken]);
```